### PR TITLE
[AMORO-2665] Replace deperated classes under ams.server by classes of ams.api

### LINKED ISSUE #2665: [Subtask]: Replace deperated classes under ams.server by classes of ams.api
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Including:

1. ServerIdentifier
2. StateField
3. TableConfiguration(and other Configuration classes)

### Parent issue

#2596

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by majin1102**
## Why are the changes needed?

Close #2665.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 1]
```diff
@@ -18,45 +18,52 @@
 
 package com.netease.arctic.api.config;
 
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:22]: We may need to import the shaded ones.

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 2]
```diff
@@ -18,45 +18,52 @@
...
+  @JsonProperty(defaultValue = TableProperties.DATA_EXPIRATION_DATE_NUMBER_FORMAT_DEFAULT)
   private String numberDateFormat;
-  // data-expire.since
-  private Since since;
+  // data-expire.base-on-rule
+  @JsonProperty(defaultValue = TableProperties.DATA_EXPIRATION_BASE_ON_RULE_DEFAULT)
+  private BaseOnRule baseOnRule;
 
-  @VisibleForTesting
+  @com.google.common.annotations.VisibleForTesting
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:60]: We may not use the full class name here.

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 3]
```diff
@@ -18,45 +18,52 @@
...
   public enum ExpireLevel {
     PARTITION,
     FILE;
 
     public static ExpireLevel fromString(String level) {
-      Preconditions.checkArgument(null != level, "Invalid level type: null");
+      com.google.common.base.Preconditions.checkArgument(null != level, "Invalid level type: null");
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:66]: We may not use the full class name here.

File: ams/api/src/main/java/com/netease/arctic/api/config/OptimizingConfig.java [hunk 4]
```diff
@@ -19,51 +19,76 @@
 package com.netease.arctic.api.config;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.base.Objects;
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/OptimizingConfig.java:22]: We may need to import the shaded ones.

File: ams/api/src/main/java/com/netease/arctic/api/config/StructuredOptionsSplitter.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.api.config;
+
+import static com.google.common.base.Preconditions.checkNotNull;
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/StructuredOptionsSplitter.java:21]: We may need to import the shaded ones.

File: ams/api/src/main/java/com/netease/arctic/api/config/TableConfiguration.java [hunk 6]
```diff
@@ -19,67 +19,29 @@
 package com.netease.arctic.api.config;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.netease.arctic.api.Action;
-import org.apache.iceberg.relocated.com.google.common.base.Objects;
+import com.google.common.base.Objects;
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/TableConfiguration.java:22]: We may need to import the shaded ones.

File: ams/api/src/main/java/com/netease/arctic/api/config/TagConfiguration.java [hunk 7]
```diff
@@ -18,36 +18,56 @@
 
 package com.netease.arctic.api.config;
 
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.base.Objects;
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/TagConfiguration.java:21]: We may need to import the shaded ones.

File: ams/server/src/main/java/com/netease/arctic/server/catalog/CatalogBuilder.java [hunk 8]
```diff
@@ -18,15 +18,11 @@
 
 package com.netease.arctic.server.catalog;
 
-import static com.netease.arctic.properties.CatalogMetaProperties.CATALOG_TYPE_AMS;
-import static com.netease.arctic.properties.CatalogMetaProperties.CATALOG_TYPE_CUSTOM;
-import static com.netease.arctic.properties.CatalogMetaProperties.CATALOG_TYPE_GLUE;
-import static com.netease.arctic.properties.CatalogMetaProperties.CATALOG_TYPE_HADOOP;
-import static com.netease.arctic.properties.CatalogMetaProperties.CATALOG_TYPE_HIVE;
+import static com.netease.arctic.properties.CatalogMetaProperties.*;
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/catalog/CatalogBuilder.java:21]: We should avoid importing the whole package.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/DashboardServer.java [hunk 9]
```diff
@@ -18,12 +18,9 @@
 
 package com.netease.arctic.server.dashboard;
 
-import static io.javalin.apibuilder.ApiBuilder.delete;
-import static io.javalin.apibuilder.ApiBuilder.get;
-import static io.javalin.apibuilder.ApiBuilder.path;
-import static io.javalin.apibuilder.ApiBuilder.post;
-import static io.javalin.apibuilder.ApiBuilder.put;
+import static io.javalin.apibuilder.ApiBuilder.*;
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/DashboardServer.java:21]: We should avoid importing the whole package.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/utils/AmsUtil.java [hunk 10]
```diff
@@ -18,16 +18,11 @@
 
 package com.netease.arctic.server.dashboard.utils;
 
-import static com.netease.arctic.server.ArcticManagementConf.HA_CLUSTER_NAME;
-import static com.netease.arctic.server.ArcticManagementConf.HA_ENABLE;
-import static com.netease.arctic.server.ArcticManagementConf.HA_ZOOKEEPER_ADDRESS;
-import static com.netease.arctic.server.ArcticManagementConf.OPTIMIZING_SERVICE_THRIFT_BIND_PORT;
-import static com.netease.arctic.server.ArcticManagementConf.SERVER_EXPOSE_HOST;
-import static com.netease.arctic.server.ArcticManagementConf.TABLE_SERVICE_THRIFT_BIND_PORT;
+import static com.netease.arctic.server.ArcticManagementConf.*;
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/utils/AmsUtil.java:21]: We should avoid importing the whole package.

File: ams/server/src/main/java/com/netease/arctic/server/terminal/kyuubi/KyuubiTerminalSessionFactory.java [hunk 11]
```diff
@@ -18,17 +18,14 @@
 
 package com.netease.arctic.server.terminal.kyuubi;
 
-import static org.apache.kyuubi.jdbc.hive.JdbcConnectionParams.AUTH_KERBEROS_AUTH_TYPE;
-import static org.apache.kyuubi.jdbc.hive.JdbcConnectionParams.AUTH_KERBEROS_AUTH_TYPE_FROM_SUBJECT;
-import static org.apache.kyuubi.jdbc.hive.JdbcConnectionParams.AUTH_PRINCIPAL;
-import static org.apache.kyuubi.jdbc.hive.JdbcConnectionParams.AUTH_USER;
+import static org.apache.kyuubi.jdbc.hive.JdbcConnectionParams.*;
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/terminal/kyuubi/KyuubiTerminalSessionFactory.java:21]: We should avoid importing the whole package.

File: ams/server/src/test/java/com/netease/arctic/server/optimizing/TestOptimizingQueue.java [hunk 12]
```diff
@@ -18,15 +18,7 @@
...
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.GROUP_TAG;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_EXECUTING_TABLES;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_EXECUTING_TASKS;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_MEMORY_BYTES_ALLOCATED;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_OPTIMIZER_INSTANCES;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_PENDING_TABLES;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_PENDING_TASKS;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_PLANING_TABLES;
-import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.OPTIMIZER_GROUP_THREADS;
+import static com.netease.arctic.server.optimizing.OptimizerGroupMetrics.*;
```
Comment (zhoujinsong) [inline ams/server/src/test/java/com/netease/arctic/server/optimizing/TestOptimizingQueue.java:21]: We should avoid importing the whole package.

File: ams/server/src/test/java/com/netease/arctic/server/optimizing/flow/CompleteOptimizingFlow.java [hunk 13]
```diff
@@ -18,27 +18,24 @@
 
 package com.netease.arctic.server.optimizing.flow;
 
-import static com.netease.arctic.table.TableProperties.SELF_OPTIMIZING_FRAGMENT_RATIO;
-import static com.netease.arctic.table.TableProperties.SELF_OPTIMIZING_MAJOR_TRIGGER_DUPLICATE_RATIO;
-import static com.netease.arctic.table.TableProperties.SELF_OPTIMIZING_MINOR_TRIGGER_FILE_CNT;
-import static com.netease.arctic.table.TableProperties.SELF_OPTIMIZING_TARGET_SIZE;
+import static com.netease.arctic.table.TableProperties.*;
```
Comment (zhoujinsong) [inline ams/server/src/test/java/com/netease/arctic/server/optimizing/flow/CompleteOptimizingFlow.java:21]: We should avoid importing the whole package.

File: mixed-format/hive/src/main/java/com/netease/arctic/hive/utils/UpgradeHiveTableUtil.java [hunk 14]
```diff
@@ -18,19 +18,15 @@
 
 package com.netease.arctic.hive.utils;
 
-import static com.netease.arctic.table.TableProperties.BASE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.CHANGE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.DEFAULT_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.FILE_FORMAT_ORC;
-import static com.netease.arctic.table.TableProperties.FILE_FORMAT_PARQUET;
+import static com.netease.arctic.table.TableProperties.*;
```
Comment (zhoujinsong) [inline mixed-format/hive/src/main/java/com/netease/arctic/hive/utils/UpgradeHiveTableUtil.java:21]: We should avoid importing the whole package.

File: mixed-format/spark/v3.3/spark/src/test/java/com/netease/arctic/spark/writer/TestMixedWriter.java [hunk 15]
```diff
@@ -18,14 +18,12 @@
 
 package com.netease.arctic.spark.writer;
 
-import static com.netease.arctic.table.TableProperties.BASE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.CHANGE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.DEFAULT_FILE_FORMAT;
+import static com.netease.arctic.table.TableProperties.*;
```
Comment (zhoujinsong) [inline mixed-format/spark/v3.3/spark/src/test/java/com/netease/arctic/spark/writer/TestMixedWriter.java:21]: We should avoid importing the whole package.

File: mixed-format/spark/v3.2/spark/src/test/java/com/netease/arctic/spark/writer/TestMixedWriter.java [hunk 16]
```diff
@@ -18,14 +18,12 @@
 
 package com.netease.arctic.spark.writer;
 
-import static com.netease.arctic.table.TableProperties.BASE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.CHANGE_FILE_FORMAT;
-import static com.netease.arctic.table.TableProperties.DEFAULT_FILE_FORMAT;
+import static com.netease.arctic.table.TableProperties.*;
```
Comment (zhoujinsong) [inline mixed-format/spark/v3.2/spark/src/test/java/com/netease/arctic/spark/writer/TestMixedWriter.java:21]: We should avoid importing the whole package.

**Review by zhoujinsong — COMMENTED** [review]
Thanks a lot for the work.
I left some comments, PTAL.

**Comment by majin1102** [comment]
> OptimizerGroupMetrics

work all done

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 17]
```diff
@@ -66,14 +75,15 @@ public static ExpireLevel fromString(String level) {
...
-    LATEST_SNAPSHOT,
-    CURRENT_TIMESTAMP;
+  public enum BaseOnRule {
+    LAST_COMMIT_TIME,
+    CURRENT_TIME;
 
-    public static Since fromString(String since) {
-      Preconditions.checkArgument(null != since, "data-expire.since is invalid: null");
+    public static BaseOnRule fromString(String since) {
+      com.google.common.base.Preconditions.checkArgument(
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:83]: We may not use the full class name here.

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 18]
```diff
@@ -95,14 +105,111 @@ public DataExpirationConfig(
...
+    this.baseOnRule = baseOnRule;
+  }
+
+  public DataExpirationConfig(ArcticTable table) {
+    Map<String, String> properties = table.properties();
+    expirationField =
+        CompatiblePropertyUtil.propertyAsString(
+            properties, TableProperties.DATA_EXPIRATION_FIELD, null);
+    Types.NestedField field = table.schema().findField(expirationField);
+    com.google.common.base.Preconditions.checkArgument(
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:124]: We may not use the full class name here.

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 19]
```diff
@@ -179,11 +286,11 @@ public boolean equals(Object o) {
     DataExpirationConfig config = (DataExpirationConfig) o;
     return enabled == config.enabled
         && retentionTime == config.retentionTime
-        && Objects.equal(expirationField, config.expirationField)
+        && com.google.common.base.Objects.equal(expirationField, config.expirationField)
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:289]: We may not use the full class name here.

File: ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java [hunk 20]
```diff
@@ -179,11 +286,11 @@ public boolean equals(Object o) {
...
         && expirationLevel == config.expirationLevel
-        && Objects.equal(dateTimePattern, config.dateTimePattern)
-        && Objects.equal(numberDateFormat, config.numberDateFormat)
-        && since == config.since;
+        && com.google.common.base.Objects.equal(dateTimePattern, config.dateTimePattern)
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/api/config/DataExpirationConfig.java:291]: We may not use the full class name here.

**Review by zhoujinsong — COMMENTED** [review]
I found some other places that need improvement. PTAL.

**Comment by majin1102** [comment]
> I found some other places that need improvement. PTAL.

done

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**